core/src/Revolution/modElement.php

```diff
@@ -1221,4 +1431,145 @@ private function getStaticFileName()
         }
         return $filename;
     }
+
+    /* -- Proposed new methods/rewrites -- */
+
+    /**
+     * Copies the needed properties from this class to the modMediaSource instance
+     */
+    public function relayStaticPropertiesToMediaSource(array $config)
+    {
+        foreach ($config as $property) {
+            $this->_source->$property = $this->$property;
+        }
+    }
+
+    /**
+     * Checks the proposed $path against a pre-defined array of protected directories.
+     * Supercedes isStaticSourceValidPath()
+     */
+    public function pathIsProtected($path): bool
+    {
+        $protectedPaths = [
+            rtrim($this->xpdo->getOption('core_path', null, MODX_CORE_PATH) . 'config' . DIRECTORY_SEPARATOR, DIRECTORY_SEPARATOR)
+        ];
+        if ($path) {
+            $sourceDirectory = rtrim(dirname($path), DIRECTORY_SEPARATOR);
+            if (in_array($sourceDirectory, $protectedPaths)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Determines whether the given file exists or can be created, as well
+     * as whether its extension is valid based on its destination
+     */
+    public function validateStaticFile($source = null)
+    {
+        $file = $this->staticFileAbsolutePath;
+
+        if ($file) {
+            if ($this->pathIsProtected($file)) {
+                return ['msgLexKey' => 'element_static_source_protected_invalid'];
+            }
+            if (file_exists($file)) {
+                if ($this->staticPathIsAbsolute) {
+                    $overwriteExistingFileOnCreate = !(bool)$this->xpdo->getOption('upload_file_exists', null, true);
+                    if ($this->isNew() && !$overwriteExistingFileOnCreate) {
+                        return ['msgLexKey' => 'element_err_staticfile_exists'];
+                    }
+                }
+                if (!is_writable($file)) {
+                    return ['msgLexKey' => 'element_static_source_immutable'];
+                }
+            } else {
+                $fileFragments = explode(DIRECTORY_SEPARATOR, $file);
+                $fileName = array_pop($fileFragments);
+                $fileDirectory = implode(DIRECTORY_SEPARATOR, $fileFragments);
+
+                if (!$this->staticPathIsWritable($fileDirectory)) {
+                    // $this->xpdo->log(
+                    //     modX::LOG_LEVEL_ERROR,
+                    //     "\rmodElement::validateStaticFile dir not writable!
+                    //     dir: " . $fileDirectory
+                    // );
+                    return ['msgLexKey' => 'file_folder_err_perms'];
+                }
+                if (!$this->_source->checkFileType($fileName)) {
+                    return [
+                        'msgLexKey' => 'file_err_ext_not_allowed',
+                        'msgData' => [ 'ext' => array_pop(explode('.', $fileName)) ]
```

Comment:
Only variables can be passed by reference. You need to introduce a variable here and pass that to `array_pop`.

Reply:
Got it, this will be changing in another way anyway.